src/main/java/gg/solarmc/kitpvp/kill/KillDataHandler.java

```diff
@@ -2,55 +2,116 @@
-            killed.getSolarPlayer().getData(KitPvpKey.INSTANCE).addDeaths(transaction,1);
+            //FUN functionality
 
-            //add assists
-            for (Player assister : assisters) {
-                assister.getSolarPlayer().getData(KitPvpKey.INSTANCE).addAssists(transaction,1);
-                killer.getSolarPlayer().getData(CreditsKey.INSTANCE).depositBalance(transaction, BigDecimal.valueOf(config.assistMoney()));
+            for (UUID assisterID : assisters) {
+                try { //blocking ""safely""
+                    SolarPlayer ass = center.lookupPlayer(assisterID).get().orElseThrow(() -> new IllegalStateException("It is not present!"));
+
+                    ass.getData(KitPvpKey.INSTANCE).addAssists(transact,1);
+                } catch (InterruptedException | ExecutionException e) {
+                    throw new UncheckedAwaitExeception(e); //handle
+                }
             }
 
-            //increment killer's killstreak
-            //reset killed's killstreak
+        }));
 
-            killer.getSolarPlayer().getData(CreditsKey.INSTANCE).depositBalance(transaction, BigDecimal.valueOf(config.killMoney()));
-        });
     }
 
-    CentralisedFuture<?> handleDeath(Player killed, Set<Player> assisters) {
-        return center.runTransact(transaction -> {
-            killed.getSolarPlayer().getData(KitPvpKey.INSTANCE).addDeaths(transaction,1);
+    CentralisedFuture<?> handleDeath(UUID killedId, Set<UUID> assisters) {
 
-            //add assists
-            for (Player assister : assisters) {
-                assister.getSolarPlayer().getData(KitPvpKey.INSTANCE).addAssists(transaction,1);
-                assister.getSolarPlayer().getData(CreditsKey.INSTANCE).depositBalance(transaction, BigDecimal.valueOf(config.assistMoney()));
-            }
 
-            //reset killed's killstreak
-        });
+        CentralisedFuture<Optional<SolarPlayer>> killedFuture = center.lookupPlayer(killedId);
+
+        return killedFuture
+                .thenApply(optional -> optional.orElseThrow(() -> new IllegalStateException("Killed player is missing!")))
+                .thenCompose(man -> center.runTransact(transact -> {
+                    man.getData(KitPvpKey.INSTANCE).addDeaths(transact,1);
+
+                    for (UUID assisterId : assisters) {
+                        //blockig time!
+                        try { //blocking ""safely""
+                            SolarPlayer ass = center.lookupPlayer(assisterId).get().orElseThrow(() -> new IllegalStateException("It is not present!"));
+
+                            ass.getData(KitPvpKey.INSTANCE).addAssists(transact,1);
+                        } catch (InterruptedException | ExecutionException e) {
+                            throw new UncheckedAwaitExeception(e); //handle
+                        }
+                    }
+                }));
+    }
+
+    public static final class CombinedPlayers {
```

Comment:
This would be a great place to use a record. Not only that, you could make the record local to your method:
```java
private void myMethod() {
  record CombinedPlayers(SolarPlayer killer, SolarPlayer killed) {}
  // do stuff using CombinedPlayers
}

// CombinedPlayers cannot be used elsewhere
```

Reply:
Records are good, but i'm totally not used to them yet